src/pages/CourseContent/index.js

```diff
@@ -37,25 +37,57 @@ function CourseContent(props) {
   }, [dispatch, courseId]);
 
   useEffect(() => {
-    const firstExercise = get(data, "exerciseList[0]");
+    let defaultExercise, defaultExerciseIndex;
+    let exerciseIdFromParams = getExerciseIdFromUrl();
+    const firstExercise = get(data, 'exerciseList[0]');
+
+    // exercises loaded
     if (firstExercise) {
-      const selectedExerciseInfo = { exercise: firstExercise, index: 0 };
-      dispatch(courseActions.updateSelectedExercise(selectedExerciseInfo));
+
+        // set default exercise as per the exercise id present in the url
+        // TBD: Ideally, when navigating to a course content page, the first exercise should be pre-computed and the url should be course/:courseId/exercise/:exerciseId so we can always use the exerciseId from the params to set the default exercise.
+        if (exerciseIdFromParams) {
+            const exerciseFromParams = data.exerciseList.find((exercise) => {
+                return exercise.id === exerciseIdFromParams;
+            });
+            if (exerciseFromParams) {
+                defaultExercise = exerciseFromParams;
+                defaultExerciseIndex = data.exerciseList.findIndex((exercise) => {
+                    return exercise.id === exerciseIdFromParams;
+                });
+            }
+        }
+
+        // exerciseId not in params or exerciseId in params not there in exercise list (eg: invalid exercise id in url)
+        if (!defaultExercise) {
```

Comment:
I think we can put this info when we are declaring `defaultExercise` and `defaultExerciseIndex`. What do you think? one less check. 🙂

Reply:
agreed!